### PR TITLE
Add custom wavelength

### PR DESCRIPTION
**Pull request by thomasculino**
This PR aims to add the ability to add a custom wavelength as a vertical line on the spectroscopy plot as you can see below:
Single_Custom_Wavelength

For now, it only supports one wavelength. Let me know if I should work on adding multiple custom wavelengths. 

I was also thinking that we should change the row with all the sliders because they do not work properly

### CODE REVIEW

File: skyportal/plot.py [hunk 1]
```diff
@@ -2329,12 +2329,36 @@ def make_spectrum_layout(obj, spectra, user, device, width, smoothing, smooth_nu
...
+        mode='float',
+    )
 
     # Track elements that need to be shifted with change in z / v
     shifting_elements = []
     renderers = []
     obj_redshift = 0 if obj.redshift is None else obj.redshift
 
+    flux_values = list(np.linspace(ymin, ymax, 100))
+    flux_values[-1] = np.nan
```
Comment (guynir42) [inline skyportal/plot.py:2344]: Putting NaN in the last value makes the whole line disappear?

Comment (guynir42) [reply skyportal/plot.py:2344]: I'm not sure where `flux_values` is used?

File: skyportal/plot.py [hunk 2]
```diff
@@ -2386,6 +2410,29 @@ def make_spectrum_layout(obj, spectra, user, device, width, smoothing, smooth_nu
...
+    w_input.js_on_change(
+        'value',
+        CustomJS(
+            args={'input': w_input, 'model_dict': model_dict},
+            code="""
+                    model_dict['custom_line'].data_source.data['x'][0] = input.value;
+                    model_dict['custom_line'].visible = true;
+                    model_dict['custom_line'].data_source.data['wavelength'][0] = input.value;
+                    model_dict['custom_line'].data_source.change.emit();
+                """,
```
Comment (guynir42) [inline skyportal/plot.py:2426]: Is there a way to make the line not visible?

**Review by guynir42 — COMMENTED** [review]
I have a few small questions.

File: skyportal/plot.py (hunk 2 above)
Comment (thomasculino) [reply skyportal/plot.py:2426]: you mean on the frontend ?

Comment (guynir42) [reply skyportal/plot.py:2426]: Yeah. After I put in a value and see the line, can I make it disappear again or do I have to refresh the page?

File: skyportal/plot.py (hunk 1 above)
Comment (thomasculino) [reply skyportal/plot.py:2344]: I copied and pasted what was below to make it work but I ended up removing flux_values from el_data, so I should move it back where it was, above the for loop line 2362.

File: skyportal/plot.py (hunk 2 above)
Comment (thomasculino) [reply skyportal/plot.py:2426]: Currently, there is no button to make this line invisible but I can add it

Comment (guynir42) [reply skyportal/plot.py:2426]: I would advocate that if the user puts in an empty value it should make the line disappear.

File: skyportal/plot.py (hunk 1 above)
Comment (guynir42) [reply skyportal/plot.py:2344]: OK I guess I just had to expand the view a little more to see it. 
Also I'm not clear on what a NaN on the last value does.

File: skyportal/plot.py (hunk 2 above)
Comment (thomasculino) [reply skyportal/plot.py:2426]: ok I will try this

**Comment by mcoughlin** [comment]
@thomasculino Any progress on this?

**Comment by thomasculino** [comment]
> @thomasculino Any progress on this?

@mcoughlin no, it still works for one wavelength but not for multiple ones.

**Comment by thomasculino** [comment]
@mcoughlin I think it is a good idea to open that issue

**Comment by mcoughlin** [comment]
@guynir42 Can we get this merged just with the one line done? Then we can open an issue?

**Comment by guynir42** [comment]
@mcoughlin one line is better than none. This is very useful as-is. Multiple lines should be a separate PR.

**Comment by mcoughlin** [comment]
@thomasculino Did you address the nan question?

**Comment by guynir42** [comment]
@thomasculino My only outstanding issue is how to remove that line if I don't want it anymore. I would expect if I input an empty value it should make it disappear.

**Comment by thomasculino** [comment]
@guynir42 I'm going to take a look at this again

**Comment by thomasculino** [comment]
> @thomasculino Did you address the nan question?

The nan was already there in the code and in the end I didn't touch this part. Honestly, I don't really know its purpose

File: skyportal/plot.py (hunk 2 above)
Comment (thomasculino) [reply skyportal/plot.py]: I followed your advice and it's working well

**Comment by guynir42** [comment]
@thomasculino let me know when ready for a last review.

**Review by guynir42 — APPROVED** [review]
Great!

**Comment by mcoughlin** [comment]
@thomasculino ready to merge?

**Comment by thomasculino** [comment]
> @thomasculino ready to merge?

@mcoughlin yes!

**Pull request merged**